javascript-carousel/main.js

```diff
@@ -0,0 +1,109 @@
+var $image = document.querySelector('img');
+var $dots = document.querySelectorAll('.fa-circle');
+var urlArray = ['images/001.png', 'images/004.png', 'images/007.png', 'images/025.png', 'images/039.png'];
+var urlArrayLength = urlArray.length;
+var entryId = setInterval(nextImage, 3000);
+var $transitionDiv = document.querySelector('.transition-div');
+var timeouts = [];
+
+function swapImage(event) {
+  if (event.target.tagName !== 'I') {
+    return;
+  }
+  var imageId = $image.getAttribute('data-id');
+  if (event.target.className === 'fas fa-chevron-left fa-5x') {
+    if (imageId === '0') {
+      $image.setAttribute('data-id', urlArrayLength - 1);
+      setTimeout(function () { $image.setAttribute('src', urlArray[urlArrayLength - 1]); }, 100);
+      $dots[(urlArrayLength - 1).toString()].className = 'fas fa-circle fa-2x padding';
+      $dots[0].className = 'far fa-circle fa-2x padding';
+    } else {
+      $image.setAttribute('data-id', Number(imageId) - 1);
+      setTimeout(function () { $image.setAttribute('src', urlArray[Number(imageId) - 1]); }, 100);
+      $dots[Number(imageId) - 1].className = 'fas fa-circle fa-2x padding';
+      $dots[Number(imageId)].className = 'far fa-circle fa-2x padding';
+    }
+  } else if (event.target.className === 'fas fa-chevron-right fa-5x') {
+    if (imageId === (urlArrayLength - 1).toString()) {
+      $image.setAttribute('data-id', 0);
+      setTimeout(function () { $image.setAttribute('src', urlArray[0]); }, 100);
+      $dots[0].className = 'fas fa-circle fa-2x padding';
+      $dots[(urlArrayLength - 1).toString()].className = 'far fa-circle fa-2x padding';
+    } else {
+      $image.setAttribute('data-id', Number(imageId) + 1);
+      setTimeout(function () { $image.setAttribute('src', urlArray[Number(imageId) + 1]); }, 100);
+      $dots[Number(imageId) + 1].className = 'fas fa-circle fa-2x padding';
+      $dots[Number(imageId)].className = 'far fa-circle fa-2x padding';
+    }
```

Comment:
This seems to have a lot of repeated code with the only difference being which index the element is that it is affecting. You should make a function with a parameter for the index to affect and just call that function.